src/ui/components/shared/WorkspaceSettingsModal/WorkspaceSubscription.tsx

```diff
@@ -187,7 +192,9 @@ const getValue = (form: HTMLFormElement, field: string) => {
 
 function AddPaymentMethod({ onDone, workspaceId }: { onDone: () => void; workspaceId: string }) {
   const [saving, setSaving] = useState(false);
-  const [error, setError] = useState<string>();
+  const [error, setError] = useState<string | undefined>(
+    stripeLoadFailed ? "Unable to add a payment method at this time." : undefined
```

Comment:
If the component is rendered before the `stripePromise` is rejected, `error` won't get updated, so no error will be shown in the UI (only the console will show "Failed to load stripe").

Reply:
:+1: I've pushed the promise handling into an effect which I think should address this but I wasn't able to test this specific scenario well.